#114 Add asdf ruby version to ruby version prompt

jschank opened the pull request:
Followed established pattern for chruby and rbenv, adding the ruby
version number from asdf to the ruby prompt, only when the local
ruby version differs from the system ruby.

Note that asdf tracks its versions in a dot file (.tool-versions) so
the concept of what is global/system vs what is local simply depends
upon what file asdf is using to determine the correct version.
Fortunately, asdf reports the .tool-versions file when querying for
the current version.

The downside to this is that correct behavior of this patch will
depend upon the format asdf uses for their provenance string.

See https://github.com/asdf-vm/asdf for details about asdf

bobthecow commented:
Thanks!

bobthecow commented:
I reworked the implementation a bit for performance, and added a comment about the asdf provenance format caveat. Mind giving it a shot and confirming that it all works as expected?

jschank commented:
Hmm, I get an exception thrown, when I use bobthefish master branch.

Lemme read about read…. :)

looks like this.


read: Variable name '2.4.2 (set by /Users/jschank/.tool-versions)' is not valid. See `help identifiers`.
~/.config/fish/functions/fish_prompt.fish (line 608):
    read -l asdf_ruby_version asdf_provenance -- $asdf_current_ruby
    ^
in function '__bobthefish_prompt_rubies'
called on standard input

in function 'iterm2_fish_prompt'
called on line 61 of file ~/.iterm2_shell_integration.fish

in command substitution
called on line 59 of file ~/.iterm2_shell_integration.fish

in command substitution
called on line 59 of file ~/.iterm2_shell_integration.fish

in function 'fish_prompt'
called on standard input

in command substitution
called on standard input


       readread - read line of input into variables
        -

   Synopsis
       read [OPTIONS] [VARIABLES...]

read: Type 'help read' for related documentation

On Dec 24, 2017, 2:31 PM -0500, Justin Hileman <notifications@github.com>, wrote:
> I reworked the implementation a bit for performance, and added a comment about the asdf provenance format caveat. Mind giving it a shot and confirming that it all works as expected?
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

jschank commented:
I think I see.
`$ asdf current ruby`
will yield this:
`2.4.2 (set by /Users/jschank/.tool-versions)`

I think read is attempting to create a variable called '2.4.2', which seems wrong.

bobthecow commented:
oh, yep. totally my fault. i tested with a fake (`foo bar`) which didn't catch this. lemme fix really quick :)

bobthecow commented:
Ok. Should be fixed by d8d7a544. And that one's even more concise!

jschank commented:
Just pulled and tested, Seems to be working fine!

bobthecow commented:
Thanks for checking it out. And thanks for your help with this :)
